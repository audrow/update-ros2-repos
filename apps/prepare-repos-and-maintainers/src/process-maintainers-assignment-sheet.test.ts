import path from 'path'
import {
  getRepoFromString,
  processMaintainersAssignmentSheet,
} from './process-maintainers-assignment-sheet'

jest.setTimeout(10000)

describe('processMaintainersAssignmentSheet', () => {
  it('should return repos', async () => {
    const maintainersAssignmentSheetPath = path.join(
      __dirname,
      '..',
      'data',
      '2022-09-14-maintainers-assignment.csv',
    )
    const reposColumnLetter = 'a'
    const maintainersStartColumnLetter = 'n'
    const headingRowNumber = 2
    const isCheckUrls = false
    const githubUrlPrefix = 'https://github.com/'

    const {repos, maintainerIds} = await processMaintainersAssignmentSheet({
      path: maintainersAssignmentSheetPath,
      headingRowNumber,
      maintainersStartColumnLetter,
      reposColumnLetter,
      githubUrlPrefix,
      isCheckUrls,
    })
    expect(repos).toMatchSnapshot()
    expect(maintainerIds).toMatchSnapshot()
  })
})

describe('getRepoFromString', () => {
  it('gets a repo that exists', async () => {
    const repoString = 'ros2/rclcpp'
    const githubUrlPrefix = 'https://github.com/'
    const isCheckUrl = true
    const repoFromUrl = await getRepoFromString(
      repoString,
      githubUrlPrefix,
      isCheckUrl,
    )
    expect(repoFromUrl).toMatchInlineSnapshot(`
      {
        "name": "rclcpp",
        "org": "ros2",
        "url": "https://github.com/ros2/rclcpp",
        "validUrl": true,
      }
    `)
  })
  it("can't find url for a repo that doesn't exist", async () => {
    const repoString = 'ros2/not-a-repo'
    const githubUrlPrefix = 'https://github.com/'
    const isCheckUrl = true
    const repoFromUrl = await getRepoFromString(
      repoString,
      githubUrlPrefix,
      isCheckUrl,
    )
    expect(repoFromUrl).toMatchInlineSnapshot(`
      {
        "name": "not-a-repo",
        "org": "ros2",
        "url": "https://github.com/ros2/not-a-repo",
        "validUrl": false,
      }
    `)
  })
  it('can skip checking a URL', async () => {
    const repoString = 'ros2/rclcpp'
    const githubUrlPrefix = 'https://github.com/'
    const isCheckUrl = false
    const repoFromUrl = await getRepoFromString(
      repoString,
      githubUrlPrefix,
      isCheckUrl,
    )
    expect(repoFromUrl).toMatchInlineSnapshot(`
      {
        "name": "rclcpp",
        "org": "ros2",
        "url": "https://github.com/ros2/rclcpp",
        "validUrl": "not checked",
      }
    `)
  })
})
