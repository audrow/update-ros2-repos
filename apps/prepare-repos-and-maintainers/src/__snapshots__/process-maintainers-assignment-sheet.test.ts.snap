// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`processMaintainersAssignmentSheet should return repos 1`] = `
[
  {
    "maintainers": [
      "wjwwood",
      "ivanpauno",
      "hidmic",
    ],
    "name": "rclcpp",
    "org": "ros2",
    "url": "https://github.com/ros2/rclcpp",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "sloretz",
      "adityapande-1995",
    ],
    "name": "rclpy",
    "org": "ros2",
    "url": "https://github.com/ros2/rclpy",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "clalancette",
      "codebot",
    ],
    "name": "ros2",
    "org": "ros2",
    "url": "https://github.com/ros2/ros2",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "ahcorde",
    ],
    "name": "rviz",
    "org": "ros2",
    "url": "https://github.com/ros2/rviz",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "clalancette",
      "audrow",
    ],
    "name": "ros2_documentation",
    "org": "ros2",
    "url": "https://github.com/ros2/ros2_documentation",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "wjwwood",
      "adityapande-1995",
      "methylDragon",
    ],
    "name": "launch",
    "org": "ros2",
    "url": "https://github.com/ros2/launch",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "gbiggs",
      "audrow",
    ],
    "name": "ros2cli",
    "org": "ros2",
    "url": "https://github.com/ros2/ros2cli",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "clalancette",
      "ahcorde",
    ],
    "name": "geometry2",
    "org": "ros2",
    "url": "https://github.com/ros2/geometry2",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "sloretz",
      "adityapande-1995",
      "quarkytale",
      "methylDragon",
    ],
    "name": "rosidl",
    "org": "ros2",
    "url": "https://github.com/ros2/rosidl",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "ivanpauno",
    ],
    "name": "rmw_cyclonedds",
    "org": "ros2",
    "url": "https://github.com/ros2/rmw_cyclonedds",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "mjeronimo",
    ],
    "name": "ament_lint",
    "org": "ament",
    "url": "https://github.com/ament/ament_lint",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "wjwwood",
      "ivanpauno",
      "audrow",
    ],
    "name": "rcl",
    "org": "ros2",
    "url": "https://github.com/ros2/rcl",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "sloretz",
      "gbiggs",
    ],
    "name": "rmw_fastrtps",
    "org": "ros2",
    "url": "https://github.com/ros2/rmw_fastrtps",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "clalancette",
      "cottsay",
      "nuclearsandwich",
    ],
    "name": "ci",
    "org": "ros2",
    "url": "https://github.com/ros2/ci",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "audrow",
    ],
    "name": "docker_images",
    "org": "osrf",
    "url": "https://github.com/osrf/docker_images",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "adityapande-1995",
      "methylDragon",
    ],
    "name": "launch_ros",
    "org": "ros2",
    "url": "https://github.com/ros2/launch_ros",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "cottsay",
    ],
    "name": "ros_buildfarm_config",
    "org": "ros2",
    "url": "https://github.com/ros2/ros_buildfarm_config",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "mjeronimo",
      "adityapande-1995",
      "audrow",
    ],
    "name": "demos",
    "org": "ros2",
    "url": "https://github.com/ros2/demos",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "quarkytale",
      "methylDragon",
    ],
    "name": "image_common",
    "org": "ros-perception",
    "url": "https://github.com/ros-perception/image_common",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "gbiggs",
      "quarkytale",
      "methylDragon",
    ],
    "name": "ros1_bridge",
    "org": "ros2",
    "url": "https://github.com/ros2/ros1_bridge",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "mjeronimo",
    ],
    "name": "ament_cmake",
    "org": "ament",
    "url": "https://github.com/ament/ament_cmake",
    "validUrl": "not checked",
  },
  {
    "maintainers": [],
    "name": "rep",
    "org": "ros-infrastructure",
    "url": "https://github.com/ros-infrastructure/rep",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "wjwwood",
      "clalancette",
      "ahcorde",
    ],
    "name": "rcutils",
    "org": "ros2",
    "url": "https://github.com/ros2/rcutils",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "tfoote",
    ],
    "name": "common_interfaces",
    "org": "ros2",
    "url": "https://github.com/ros2/common_interfaces",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "sloretz",
      "quarkytale",
    ],
    "name": "rosidl_python",
    "org": "ros2",
    "url": "https://github.com/ros2/rosidl_python",
    "validUrl": "not checked",
  },
  {
    "maintainers": [],
    "name": "robot_state_publisher",
    "org": "ros",
    "url": "https://github.com/ros/robot_state_publisher",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "wjwwood",
      "ivanpauno",
      "methylDragon",
    ],
    "name": "rmw",
    "org": "ros2",
    "url": "https://github.com/ros2/rmw",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "quarkytale",
    ],
    "name": "rqt",
    "org": "ros-visualization",
    "url": "https://github.com/ros-visualization/rqt",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "ahcorde",
      "adityapande-1995",
    ],
    "name": "examples",
    "org": "ros2",
    "url": "https://github.com/ros2/examples",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "gbiggs",
    ],
    "name": "message_filters",
    "org": "ros2",
    "url": "https://github.com/ros2/message_filters",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "methylDragon",
    ],
    "name": "system_tests",
    "org": "ros2",
    "url": "https://github.com/ros2/system_tests",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "mjeronimo",
    ],
    "name": "rqt_bag",
    "org": "ros-visualization",
    "url": "https://github.com/ros-visualization/rqt_bag",
    "validUrl": "not checked",
  },
  {
    "maintainers": [],
    "name": "urdfdom",
    "org": "ros",
    "url": "https://github.com/ros/urdfdom",
    "validUrl": "not checked",
  },
  {
    "maintainers": [],
    "name": "pluginlib",
    "org": "ros",
    "url": "https://github.com/ros/pluginlib",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "wjwwood",
    ],
    "name": "rqt_image_view",
    "org": "ros-visualization",
    "url": "https://github.com/ros-visualization/rqt_image_view",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "wjwwood",
      "clalancette",
    ],
    "name": "design",
    "org": "ros2",
    "url": "https://github.com/ros2/design",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "gbiggs",
    ],
    "name": "rcl_interfaces",
    "org": "ros2",
    "url": "https://github.com/ros2/rcl_interfaces",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "wjwwood",
      "methylDragon",
    ],
    "name": "rcpputils",
    "org": "ros2",
    "url": "https://github.com/ros2/rcpputils",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "cottsay",
    ],
    "name": "yaml_cpp_vendor",
    "org": "ros2",
    "url": "https://github.com/ros2/yaml_cpp_vendor",
    "validUrl": "not checked",
  },
  {
    "maintainers": [],
    "name": "class_loader",
    "org": "ros",
    "url": "https://github.com/ros/class_loader",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "quarkytale",
    ],
    "name": "qt_gui_core",
    "org": "ros-visualization",
    "url": "https://github.com/ros-visualization/qt_gui_core",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "cottsay",
    ],
    "name": "orocos_kdl_vendor",
    "org": "ros2",
    "url": "https://github.com/ros2/orocos_kdl_vendor",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "wjwwood",
    ],
    "name": "rmw_implementation",
    "org": "ros2",
    "url": "https://github.com/ros2/rmw_implementation",
    "validUrl": "not checked",
  },
  {
    "maintainers": [],
    "name": "rosdoc2",
    "org": "ros-infrastructure",
    "url": "https://github.com/ros-infrastructure/rosdoc2",
    "validUrl": "not checked",
  },
  {
    "maintainers": [],
    "name": "joint_state_publisher",
    "org": "ros",
    "url": "https://github.com/ros/joint_state_publisher",
    "validUrl": "not checked",
  },
  {
    "maintainers": [],
    "name": "kdl_parser",
    "org": "ros",
    "url": "https://github.com/ros/kdl_parser",
    "validUrl": "not checked",
  },
  {
    "maintainers": [],
    "name": "ros_tutorials",
    "org": "ros",
    "url": "https://github.com/ros/ros_tutorials",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "mjeronimo",
    ],
    "name": "rqt_graph",
    "org": "ros-visualization",
    "url": "https://github.com/ros-visualization/rqt_graph",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "quarkytale",
    ],
    "name": "rqt_reconfigure",
    "org": "ros-visualization",
    "url": "https://github.com/ros-visualization/rqt_reconfigure",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "wjwwood",
    ],
    "name": "osrf_pycommon",
    "org": "osrf",
    "url": "https://github.com/osrf/osrf_pycommon",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "cottsay",
    ],
    "name": "buildfarm_perf_tests",
    "org": "ros2",
    "url": "https://github.com/ros2/buildfarm_perf_tests",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "cottsay",
    ],
    "name": "libyaml_vendor",
    "org": "ros2",
    "url": "https://github.com/ros2/libyaml_vendor",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "wjwwood",
      "clalancette",
    ],
    "name": "rcl_logging",
    "org": "ros2",
    "url": "https://github.com/ros2/rcl_logging",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "methylDragon",
    ],
    "name": "realtime_support",
    "org": "ros2",
    "url": "https://github.com/ros2/realtime_support",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "sloretz",
      "quarkytale",
    ],
    "name": "rosidl_typesupport",
    "org": "ros2",
    "url": "https://github.com/ros2/rosidl_typesupport",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "sloretz",
    ],
    "name": "rosidl_typesupport_fastrtps",
    "org": "ros2",
    "url": "https://github.com/ros2/rosidl_typesupport_fastrtps",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "mjeronimo",
    ],
    "name": "rosindex",
    "org": "ros-infrastructure",
    "url": "https://github.com/ros-infrastructure/rosindex",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "gbiggs",
    ],
    "name": "angles",
    "org": "ros",
    "url": "https://github.com/ros/angles",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "gbiggs",
    ],
    "name": "bond_core",
    "org": "ros",
    "url": "https://github.com/ros/bond_core",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "ahcorde",
    ],
    "name": "geometry_tutorials",
    "org": "ros",
    "url": "https://github.com/ros/geometry_tutorials",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "gbiggs",
    ],
    "name": "resource_retriever",
    "org": "ros",
    "url": "https://github.com/ros/resource_retriever",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "quarkytale",
    ],
    "name": "interactive_markers",
    "org": "ros-visualization",
    "url": "https://github.com/ros-visualization/interactive_markers",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "sloretz",
    ],
    "name": "python_qt_binding",
    "org": "ros-visualization",
    "url": "https://github.com/ros-visualization/python_qt_binding",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "methylDragon",
    ],
    "name": "rqt_plot",
    "org": "ros-visualization",
    "url": "https://github.com/ros-visualization/rqt_plot",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "quarkytale",
    ],
    "name": "ament_index",
    "org": "ament",
    "url": "https://github.com/ament/ament_index",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "cottsay",
    ],
    "name": "performance_test",
    "org": "ros2",
    "url": "https://github.com/ros2/performance_test",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "ivanpauno",
    ],
    "name": "rmw_dds_common",
    "org": "ros2",
    "url": "https://github.com/ros2/rmw_dds_common",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "clalancette",
    ],
    "name": "depthimage_to_laserscan",
    "org": "ros-perception",
    "url": "https://github.com/ros-perception/depthimage_to_laserscan",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "methylDragon",
    ],
    "name": "rqt_topic",
    "org": "ros-visualization",
    "url": "https://github.com/ros-visualization/rqt_topic",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "quarkytale",
    ],
    "name": "ament_package",
    "org": "ament",
    "url": "https://github.com/ament/ament_package",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "cottsay",
    ],
    "name": "uncrustify_vendor",
    "org": "ament",
    "url": "https://github.com/ament/uncrustify_vendor",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "gbiggs",
    ],
    "name": "osrf_testing_tools_cpp",
    "org": "osrf",
    "url": "https://github.com/osrf/osrf_testing_tools_cpp",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "methylDragon",
    ],
    "name": "orocos_kinematics_dynamics",
    "org": "ros2",
    "url": "https://github.com/ros2/orocos_kinematics_dynamics",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "cottsay",
    ],
    "name": "pybind11_vendor",
    "org": "ros2",
    "url": "https://github.com/ros2/pybind11_vendor",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "sloretz",
    ],
    "name": "python_cmake_module",
    "org": "ros2",
    "url": "https://github.com/ros2/python_cmake_module",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "sloretz",
    ],
    "name": "rosidl_runtime_py",
    "org": "ros2",
    "url": "https://github.com/ros2/rosidl_runtime_py",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "quarkytale",
    ],
    "name": "test_interface_files",
    "org": "ros2",
    "url": "https://github.com/ros2/test_interface_files",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "clalancette",
      "ahcorde",
    ],
    "name": "urdf",
    "org": "ros2",
    "url": "https://github.com/ros2/urdf",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "gbiggs",
    ],
    "name": "filters",
    "org": "ros",
    "url": "https://github.com/ros/filters",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "nuclearsandwich",
    ],
    "name": "ros_environment",
    "org": "ros",
    "url": "https://github.com/ros/ros_environment",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "clalancette",
    ],
    "name": "urdfdom_headers",
    "org": "ros",
    "url": "https://github.com/ros/urdfdom_headers",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "cottsay",
    ],
    "name": "googletest",
    "org": "ament",
    "url": "https://github.com/ament/googletest",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "cottsay",
    ],
    "name": "google_benchmark_vendor",
    "org": "ament",
    "url": "https://github.com/ament/google_benchmark_vendor",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "audrow",
    ],
    "name": "multiarch-docker-image-generation",
    "org": "osrf",
    "url": "https://github.com/osrf/multiarch-docker-image-generation",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "ahcorde",
      "cottsay",
    ],
    "name": "ignition_cmake2_vendor",
    "org": "ignition",
    "url": "https://github.com/ignition/ignition_cmake2_vendor",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "ahcorde",
      "cottsay",
    ],
    "name": "ignition_math6_vendor",
    "org": "ignition",
    "url": "https://github.com/ignition/ignition_math6_vendor",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "mjeronimo",
    ],
    "name": "navigation_msgs",
    "org": "ros-planning",
    "url": "https://github.com/ros-planning/navigation_msgs",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "mjeronimo",
    ],
    "name": "Mimick",
    "org": "ros2",
    "url": "https://github.com/ros2/Mimick",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "methylDragon",
    ],
    "name": "ament_cmake_ros",
    "org": "ros2",
    "url": "https://github.com/ros2/ament_cmake_ros",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "nuclearsandwich",
    ],
    "name": "choco-packages",
    "org": "ros2",
    "url": "https://github.com/ros2/choco-packages",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "cottsay",
    ],
    "name": "console_bridge_vendor",
    "org": "ros2",
    "url": "https://github.com/ros2/console_bridge_vendor",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "cottsay",
    ],
    "name": "eigen3_cmake_module",
    "org": "ros2",
    "url": "https://github.com/ros2/eigen3_cmake_module",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "gbiggs",
    ],
    "name": "example_interfaces",
    "org": "ros2",
    "url": "https://github.com/ros2/example_interfaces",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "cottsay",
    ],
    "name": "mimick_vendor",
    "org": "ros2",
    "url": "https://github.com/ros2/mimick_vendor",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "ahcorde",
    ],
    "name": "performance_test_fixture",
    "org": "ros2",
    "url": "https://github.com/ros2/performance_test_fixture",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "methylDragon",
    ],
    "name": "rmw_connext",
    "org": "ros2",
    "url": "https://github.com/ros2/rmw_connext",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "quarkytale",
    ],
    "name": "ros2cli_common_extensions",
    "org": "ros2",
    "url": "https://github.com/ros2/ros2cli_common_extensions",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "quarkytale",
    ],
    "name": "ros_testing",
    "org": "ros2",
    "url": "https://github.com/ros2/ros_testing",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "mjeronimo",
    ],
    "name": "ros_workspace",
    "org": "ros2",
    "url": "https://github.com/ros2/ros_workspace",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "sloretz",
    ],
    "name": "rosidl_core",
    "org": "ros2",
    "url": "https://github.com/ros2/rosidl_core",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "sloretz",
    ],
    "name": "rosidl_dds",
    "org": "ros2",
    "url": "https://github.com/ros2/rosidl_dds",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "sloretz",
    ],
    "name": "rosidl_defaults",
    "org": "ros2",
    "url": "https://github.com/ros2/rosidl_defaults",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "sloretz",
    ],
    "name": "rosidl_typesupport_connext",
    "org": "ros2",
    "url": "https://github.com/ros2/rosidl_typesupport_connext",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "quarkytale",
    ],
    "name": "rpyutils",
    "org": "ros2",
    "url": "https://github.com/ros2/rpyutils",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "cottsay",
      "quarkytale",
    ],
    "name": "spdlog_vendor",
    "org": "ros2",
    "url": "https://github.com/ros2/spdlog_vendor",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "clalancette",
      "ahcorde",
    ],
    "name": "teleop_twist_joy",
    "org": "ros2",
    "url": "https://github.com/ros2/teleop_twist_joy",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "clalancette",
      "ahcorde",
    ],
    "name": "teleop_twist_keyboard",
    "org": "ros2",
    "url": "https://github.com/ros2/teleop_twist_keyboard",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "cottsay",
    ],
    "name": "tinyxml2_vendor",
    "org": "ros2",
    "url": "https://github.com/ros2/tinyxml2_vendor",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "cottsay",
    ],
    "name": "tinyxml_vendor",
    "org": "ros2",
    "url": "https://github.com/ros2/tinyxml_vendor",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "methylDragon",
    ],
    "name": "tlsf",
    "org": "ros2",
    "url": "https://github.com/ros2/tlsf",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "gbiggs",
    ],
    "name": "unique_identifier_msgs",
    "org": "ros2",
    "url": "https://github.com/ros2/unique_identifier_msgs",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "gbiggs",
    ],
    "name": "variants",
    "org": "ros2",
    "url": "https://github.com/ros2/variants",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "tfoote",
    ],
    "name": "rosdoc_lite",
    "org": "ros-infrastructure",
    "url": "https://github.com/ros-infrastructure/rosdoc_lite",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "quarkytale",
    ],
    "name": "laser_geometry",
    "org": "ros-perception",
    "url": "https://github.com/ros-perception/laser_geometry",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "scpeters",
    ],
    "name": "console_bridge",
    "org": "ros",
    "url": "https://github.com/ros/console_bridge",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "clalancette",
    ],
    "name": "eigen_stl_containers",
    "org": "ros",
    "url": "https://github.com/ros/eigen_stl_containers",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "clalancette",
    ],
    "name": "urdf_parser_py",
    "org": "ros",
    "url": "https://github.com/ros/urdf_parser_py",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "gbiggs",
    ],
    "name": "rqt_action",
    "org": "ros-visualization",
    "url": "https://github.com/ros-visualization/rqt_action",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "gbiggs",
    ],
    "name": "rqt_common_plugins",
    "org": "ros-visualization",
    "url": "https://github.com/ros-visualization/rqt_common_plugins",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "quarkytale",
    ],
    "name": "rqt_console",
    "org": "ros-visualization",
    "url": "https://github.com/ros-visualization/rqt_console",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "methylDragon",
    ],
    "name": "rqt_msg",
    "org": "ros-visualization",
    "url": "https://github.com/ros-visualization/rqt_msg",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "mjeronimo",
    ],
    "name": "rqt_publisher",
    "org": "ros-visualization",
    "url": "https://github.com/ros-visualization/rqt_publisher",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "quarkytale",
    ],
    "name": "rqt_py_console",
    "org": "ros-visualization",
    "url": "https://github.com/ros-visualization/rqt_py_console",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "methylDragon",
    ],
    "name": "rqt_service_caller",
    "org": "ros-visualization",
    "url": "https://github.com/ros-visualization/rqt_service_caller",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "mjeronimo",
    ],
    "name": "rqt_shell",
    "org": "ros-visualization",
    "url": "https://github.com/ros-visualization/rqt_shell",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "gbiggs",
    ],
    "name": "rqt_srv",
    "org": "ros-visualization",
    "url": "https://github.com/ros-visualization/rqt_srv",
    "validUrl": "not checked",
  },
  {
    "maintainers": [
      "cottsay",
    ],
    "name": "tango_icons_vendor",
    "org": "ros-visualization",
    "url": "https://github.com/ros-visualization/tango_icons_vendor",
    "validUrl": "not checked",
  },
]
`;

exports[`processMaintainersAssignmentSheet should return repos 2`] = `
[
  "wjwwood",
  "ivanpauno",
  "hidmic",
  "sloretz",
  "adityapande-1995",
  "clalancette",
  "codebot",
  "ahcorde",
  "audrow",
  "methylDragon",
  "gbiggs",
  "quarkytale",
  "mjeronimo",
  "cottsay",
  "nuclearsandwich",
  "tfoote",
  "scpeters",
]
`;
